crates/edr_napi/src/provider.rs

```diff
@@ -35,12 +37,24 @@ impl Provider {
         _context: &EdrContext,
         config: ProviderConfig,
         logger_config: LoggerConfig,
+        tracing_config: serde_json::Value,
         #[napi(ts_arg_type = "(event: SubscriptionEvent) => void")] subscriber_callback: JsFunction,
     ) -> napi::Result<JsObject> {
-        let config = edr_provider::ProviderConfig::try_from(config)?;
         let runtime = runtime::Handle::current();
 
-        let logger = Box::new(Logger::new(&env, logger_config)?);
+        let config = edr_provider::ProviderConfig::try_from(config)?;
+
+        let build_info_config: edr_solidity::contract_decoder::BuildInfoConfig =
+            serde_json::from_value(tracing_config)?;
+        let contract_decoder = ContractDecoder::new(&build_info_config)
+            .map_err(|error| napi::Error::from_reason(error.to_string()))?;
+        let contract_decoder = Arc::new(contract_decoder);
```

Comment:
This work should be done on a thread from the blocking thread pool to avoid blocking the JS event loop, but that's difficult to achieve, because this is not an async function (precluded by the `Env` argument not being `Send`) and the contract decoder needs to be passed in to the logger which also needs the `Env`.

Since constructing the contract decoder (previously called `VmTraceDecoder`) was blocking previously as well, and this PR is getting very large, I don't think we have to fix it here.